src/frontend/src/components/profile-userPostList.vue

```diff
@@ -0,0 +1,547 @@
+<script setup lang="ts">
+import {watch, onMounted, onBeforeUnmount, ref, computed} from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import {services} from '../services/api';
+import type {PostBo} from '../types/post'
+import { useAuthStore } from '../stores/auth-store';
+import { useNotification } from '../composables/use-notification.ts';
+import type { UserProfileBo } from '../types/user.ts';
+import { useContextMenu } from '../composables/use-context-menu';
+
+const route = useRoute();
+const router = useRouter();
+const authStore = useAuthStore();
+const notification = useNotification();
+const {
+    contextMenuItemId: contextMenuPostId,
+    contextMenuPosition,
+    showContextMenu,
+    showMobileMenu,
+    handleRightClick,
+    toggleMobileMenu,
+    closeAllMenus
+} = useContextMenu();
+
+const viewProfile = (postId: string, event?: Event) => {
+    if (event) {
+        event.stopPropagation();
+    }
+    
+    closeAllMenus();
+    
+    const post = posts.value.find(p => p.uid === postId);
+    if (post) {
+        router.push({ path: '/profile', query: { id: post.creatorUid } });
+    }
+};
```

Comment:
add a line between const and function definitions

```suggestion
};

```

Reply:
added